### PR TITLE
Add syncing script and modify docker config accordingly 

### PR DESCRIPTION
**Pull request by NateBrune**

### CODE REVIEW

File: python_scripts/tests/test_sync.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+        if( steemd_err != None ):
+         self.steemd_err = open(steemd_err, 'w')
+        else:
+         self.steemd_err = self._FNULL
+
+    def _get_config(self):
+       return "# no seed-node in config file or command line\n" \
+           + "rpc-endpoint = 127.0.0.1:8090              # bind to localhost to secure RPC API access\n" \
+           + "seed-node = seed.riversteem.com:2001       # riverhead\n" \
```
Comment (sneak) [inline python_scripts/tests/test_sync.py]: as @Gandalf-the-Grey mentioned, do not hard code seed nodes.

File: python_scripts/tests/test_sync.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+           + "seed-node = 52.74.152.79:2001              # smooth\n" \
+           + "enable-plugin = witness "+ " ".join( self.plugins ) + "\n" \
+           + "public-api = database_api login_api " + " ".join( self.apis ) + "\n"
+
+    def panic(self, stl, lbn):
+        print("\nWATCHDOG: Error syncing between block #"+str(stl)+" and #"+str(lbn))
+        steemd.kill()
+        sys.exit(1)
+
+    def watchForHangDurringSync(self, steemd):
```
Comment (sneak) [inline python_scripts/tests/test_sync.py]: `during`

File: python_scripts/tests/test_sync.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                duplicates+=1
+                if duplicates > 10:
+                    panic(second_to_last, local_block_number) # We have been on the same block for 100 seconds.
+                    return
+            else:
+                duplicates=0
+                second_to_last = last_local_block_number
+                last_irreversible_block_num = remoteProps['last_irreversible_block_num']
+                last_local_block_number = local_block_number
+                if(local_block_number>=last_irreversible_block_num):
```
Comment (sneak) [inline python_scripts/tests/test_sync.py]: pep8 this file please

File: python_scripts/tests/test_sync.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    parser.add_argument( '--log-file', '-l', type=str, required=False, help='The location of the file which will be logged to.' )
+    parser.add_argument( '--seed-nodes', '-sn', type=str, required=False, help='The location of the file which contains seed nodes.' )
+    args = parser.parse_args()
+
+    config = None
+    seeds = None
+    logs = None
+    steemd = Path( args.steemd )
+    if( not steemd.exists() ):
+       print( 'Error: steemd does not exist.' )
```
Comment (sneak) [inline python_scripts/tests/test_sync.py]: throw exceptions

**Comment by theoreticalbts** [comment]
I think this is very much an application of Steem, not an integral part of Steemd itself.  Therefore it should live outside this repo.  Either inherit the `Dockerfile` or include this repo as a submodule.

Also, this requires `steemapi`.  So it should have `requirements.txt` listing where to find it.  Ideally this should packaged as a Python package with `setup.py`.

File: python_scripts/tests/test_sync.py (hunk 1 above)
Comment (sneak) [inline python_scripts/tests/test_sync.py]: as @Gandalf-the-Grey mentioned, do not hard code seed nodes.

File: python_scripts/tests/test_sync.py (hunk 2 above)
Comment (sneak) [inline python_scripts/tests/test_sync.py]: during

File: python_scripts/tests/test_sync.py (hunk 3 above)
Comment (sneak) [inline python_scripts/tests/test_sync.py]: make sure this file passes pep8 linting please

**Comment by sneak** [comment]
@NateBrune justin made some changes that are going to require that you update this PR, sorry

**Pull request closed without merging**